x-pack/plugins/task_manager/server/buffered_task_store.ts

```diff
@@ -45,6 +61,6 @@ export class BufferedTaskStore implements Updatable {
   }
 
   public async remove(id: string): Promise<void> {
-    return this.taskStore.remove(id);
+    await unwrapPromise(this.bufferedRemove({ id }));
```

Comment:
The call to `unwrapPromise` makes this function throw errors whenever `asErr` is returned from the buffer. This makes the function work like before and throw errors if ever an error occurred (conflict, not found, etc).